torch/onnx/symbolic_opset13.py

```diff
@@ -312,3 +313,62 @@ def repeat_interleave(g, self, repeats, dim=None, output_size=None):
     loop_out = loop.node().output()
     loop_out = g.op("ConcatFromSequence", loop_out, axis_i=dim)
     return loop_out
+
+
+@parse_args("v", "i", "i", "i")
+def diagonal(g, self, offset, dim1, dim2):
+    dim1_size = size(g, self, dim=g.op("Constant", value_t=torch.LongTensor([dim1])))
+    dim2_size = size(g, self, dim=g.op("Constant", value_t=torch.LongTensor([dim2])))
+
+    # Create appropriate mask
+    mask_shape = [dim1_size, dim2_size]
+    mask_shape = g.op("Concat", *mask_shape, axis_i=0)
+    mask = zeros(g, mask_shape, None, None, None)
+    mask = g.op("EyeLike", mask, k_i=offset)
+
+    # dim1 and dim2 appended as a dimension at the end of the shape
+    rank = sym_help._get_tensor_rank(self)
+    if rank is not None:
+        axes = list(range(rank))
+        axes.remove(dim1)
+        axes.remove(dim2)
+        self = g.op("Transpose", self, perm_i=axes + [dim1, dim2])
+    else:
+        return _unimplemented("diagonal", "negative axis with unknown input rank")
+
+    # Update mask
+    axes = list(range(rank))
+    mask_shape = [size(g, self, 
+                       dim=g.op("Constant", value_t=torch.LongTensor([axis]))) for axis in axes[:-2]]
+    mask_shape.append(g.op("Constant", value_t=torch.LongTensor([-1])))
+    mask_shape.append(g.op("Constant", value_t=torch.LongTensor([-1])))
+    mask_shape = g.op("Concat", *mask_shape, axis_i=0)
+    mask = expand(g, mask, mask_shape, None)
+    
+    # Multiply input and mask
```

Comment:
Please expand the comment to explain why the multiplication is happening. Just stating that there's a multiplication is not useful.

Reply:
Added description